build/tests/service-test.sh

```diff
@@ -102,28 +102,77 @@ for v in $(seq 0 $((${port_count} - 1))) ; do
-say run unit tests for eservice database
+# these functions attemp to create a reproducible environment for
+# each of the tests
 # -----------------------------------------------------------------
+function begin_test {
+    yell $@
+    try ${PDO_SOURCE_ROOT}/build/tests/site-configuration.psh ${PSHELL_OPTS}
+}
+
+function end_test {
+    yell end test
+
+    try pdo-service-db clear ${PSHELL_OPTS}
+    try pdo-service-groups clear ${PSHELL_OPTS}
+}
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+begin_test run unit tests for eservice database
+
 try ${PDO_SOURCE_ROOT}/build/tests/service-storage-test.psh \
     ${PSHELL_OPTS} \
     --bind tmpfile ${ESDB_FILE}
 
 say create the eservice database using database CLI
 try pdo-service-db clear ${PSHELL_OPTS}
+try pdo-service-groups clear
+
 try pdo-service-db add --url http://${F_SERVICE_HOST}:7101 --name es7101 --type eservice ${PSHELL_OPTS}
 try pdo-service-db add --url http://${F_SERVICE_HOST}:7102 --name es7102 --type eservice ${PSHELL_OPTS}
 try pdo-service-db add --url http://${F_SERVICE_HOST}:7103 --name es7103 --type eservice ${PSHELL_OPTS}
 try pdo-service-db add --url http://${F_SERVICE_HOST}:7104 --name es7104 --type eservice ${PSHELL_OPTS}
 try pdo-service-db add --url http://${F_SERVICE_HOST}:7105 --name es7105 --type eservice ${PSHELL_OPTS}
 
+end_test
+
 # -----------------------------------------------------------------
-say start storage service test
-#  ----------------------------------------------------------------
+# ----------------------------------------------------------------
+begin_test start storage service test
+
 try pdo-test-storage ${LOCAL_OPTS} --url http://${F_SERVICE_HOST}:7201
 
+end_test
+
 # -----------------------------------------------------------------
-say start request test
 # -----------------------------------------------------------------
+begin_test run unit tests for service groups database
+
+try ${PDO_SOURCE_ROOT}/build/tests/service-groups-test.psh \
+    ${PSHELL_OPTS} \
+    --bind tmpfile ${GROUPS_FILE}
+
+say create the service and groups database using the groups CLI
+try pdo-service-db clear ${PSHELL_OPTS}
+try pdo-service-groups clear
+
+try pdo-service-db add --url http://${F_SERVICE_HOST}:7101 --name es7101 --type eservice ${PSHELL_OPTS}
+try pdo-service-db add --url http://${F_SERVICE_HOST}:7102 --name es7102 --type eservice ${PSHELL_OPTS}
+try pdo-service-db add --url http://${F_SERVICE_HOST}:7103 --name es7103 --type eservice ${PSHELL_OPTS}
+try pdo-service-db add --url http://${F_SERVICE_HOST}:7104 --name es7104 --type eservice ${PSHELL_OPTS}
+try pdo-service-db add --url http://${F_SERVICE_HOST}:7105 --name es7105 --type eservice ${PSHELL_OPTS}
+
+try pdo-eservice create --group test1
+try pdo-eservice add --group test1 --url http://${F_SERVICE_HOST}:7101 http://${F_SERVICE_HOST}:7102
+try pdo-eservice add --group test1 --name es7103 es7104
```

Comment:
in line 168, you are assuming that there is only one eservice-db available per environment, and you are also assuming its name ? May be pdo-eservice add --group test1 --db-name <db-name> --name es7103 es7104 ?

Reply:
you can specify group database and service database on the command line.